File: 2023/tests/day24.test.js
const fs = require('fs');
const { part1, part2 } = require('../src/day24');

const example1 = `
19, 13, 30 @ -2,  1, -2
18, 19, 22 @ -1, -1, -2
20, 25, 34 @ -2, -2, -4
12, 31, 28 @ -1, -2, -1
20, 19, 15 @  1, -5, -3
`.trim();

const input = fs.readFileSync('inputs/input24.txt', 'utf8');

describe('part 1', () => {
  test('example', () => {
    expect(part1(example1)).toBe(2);
  });

  test('solution', () => {
    expect(part1(input)).toBe(14672);
  });
});

describe('part 2', () => {
  test('example', async () => {
    expect(await part2(example1)).toBe(47);
  });

  test('solution', async () => {
    expect(await part2(input)).toBe(646810057104753);
  });
});

File: 2024/AdventOfCode/Tests/Day24Test.cs
using NUnit.Framework;

namespace AdventOfCode.Tests;

[TestFixture]
public class Day24Test {
    private const string Example = "Inputs/24-Example.txt";
    private const string Example2 = "Inputs/24-Example2.txt";
    private const string Input = "Inputs/24.txt";
    private static Day24 _day;

    [TestFixture]
    public class Part1 {
        [Test]
        public void TestExample() {
            _day = new Day24(Example);
            const string expected = "4";
            var solution = _day.Solve_1();
            Assert.That(solution.Result, Is.EqualTo(expected));
        }

        [Test]
        public void TestExample2() {
            _day = new Day24(Example2);
            const string expected = "2024";
            var solution = _day.Solve_1();
            Assert.That(solution.Result, Is.EqualTo(expected));
        }

        [Test]
        public void TestInput() {
            _day = new Day24(Input);
            const string expected = "60714423975686";
            var solution = _day.Solve_1();
            Assert.That(solution.Result, Is.EqualTo(expected));
        }
    }

    [TestFixture]
    public class Part2 {
        [Test]
        public void TestInput() {
            _day = new Day24(Input);
            const string expected = "cgh,frt,pmd,sps,tst,z05,z11,z23";
            var solution = _day.Solve_2();
            Assert.That(solution.Result, Is.EqualTo(expected));
        }
    }
}